include/RAJA/policy/tbb/reduce.hpp

```diff
@@ -66,6 +66,16 @@ class ReduceTBB
     data->local() = init_val;
   }
 
+  //There should be a better way to reset data
+  void reset(T init_val, T initializer)
+  {
+
+    data = std::shared_ptr<tbb::combinable<T>>(
+           std::make_shared<tbb::combinable<T>>([=]() { return initializer; }));
+
```

Comment:
You should be able to iterate over all the members of the combinable.

That said, have you added anything to debug the reset method? It's only safe if all children have already been destroyed and it's invoked from the single master thread right?

Reply:
Yes, you have a point thanks; I will work on that next. Something I am not quite sure about is what I need to reset in data. Is invoking clear and resetting the local value enough? I see that initializer gets used to construct data but I don't how else it gets used.

Reply:
@artv3 and @trws I don't think we have to worry about making the reset operations work when there are any children around. The use case that has come up in conversations with app folks is that they want to re-use a reduction object in multiple kernels. They would retrieve the reduced value after each kernel and reset the reduction object to a new initial value before using it again.  So reset should look just like the initialization part of (non-copy) construction. In fact, I would suggest implementing the initialization operations in the reset method and then calling that in the main constructor.

Reply:
As long as it is always used that way, it would be ok.  That said, if any asynchronous kernel using a reduction ends after a reset has been done on that reduction, the child from that kernel will invalidate the next reduction without warning, or any way to detect that a mistake has been made other than a wrong answer.  For now it probably only matters for CUDA, but maybe add a sync at the beginning of reset to match the one in `get()`?

Reply:
Yes, that is a good idea.

Reply:
Agreed 👍